src/converse-register.js

```diff
@@ -219,20 +224,37 @@
                      * Parameters:
                      *      (Submit Event) ev - Form submission event.
                      */
-                    if (ev && ev.preventDefault) { ev.preventDefault(); }
-                    var $form = $(ev.target),
-                        $domain_input = $form.find('input[name=domain]'),
-                        domain = $domain_input.val();
-                    if (!domain) {
-                        $domain_input.addClass('error');
-                        return;
+                    var cancel_label = '';
+                    if (!_converse.registration_domain) { 
+                        if (ev && ev.preventDefault) { ev.preventDefault(); }
+                        var $form = $(ev.target),
+                            $domain_input = $form.find('input[name=domain]'),
+                            domain = $domain_input.val();
+                        if (!domain) {
+                            $domain_input.addClass('error');
+                            return;
+                        }
+                        cancel_label = __('Cancel');
+                    }
+                    else {
+                        domain = _converse.registration_domain;
+                    }
+
+                    var reg_req = tpl_registration_request({
+                        cancel: cancel_label,
+                        info_message: _converse.__('Requesting a registration form from the XMPP server')
+                    });
+
+                    if (!_converse.registration_domain) {
+                        $form.find('input[type=submit]').hide()
```

Comment:
But the check here is `!_converse.registration_domain` which ensures the `submit` button is always rendered in this scope.